apm-sniffer/apm-sdk-plugin/ehcache-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/ehcache/v2/EhcacheConstructorInterceptor.java

```diff
@@ -18,18 +18,28 @@
 
 package org.apache.skywalking.apm.plugin.ehcache.v2;
 
+import net.sf.ehcache.Cache;
 import net.sf.ehcache.config.CacheConfiguration;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
 
 public class EhcacheConstructorInterceptor implements InstanceConstructorInterceptor {
     @Override
     public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
-        CacheConfiguration cacheConfiguration = (CacheConfiguration) allArguments[0];
+        try {
+            CacheConfiguration cacheConfiguration = (CacheConfiguration) allArguments[0];
 
-        // get cache name
-        if (cacheConfiguration != null) {
-            objInst.setSkyWalkingDynamicField(new EhcacheEnhanceInfo(cacheConfiguration.getName()));
+            // get cache name
+            if (cacheConfiguration != null) {
+                objInst.setSkyWalkingDynamicField(new EhcacheEnhanceInfo(cacheConfiguration.getName()));
+            }
+        } catch (ClassCastException e) {
```

Comment:
You don't need to use this `try/catch`. `getConstructorsInterceptPoints` returns an array, you should set up the interceptors separately. This one you didi has a performance issue.

Reply:
Can I use `instanseof` instead? 
```java
       if (argument instanceof CacheConfiguration) {
            CacheConfiguration cacheConfiguration = (CacheConfiguration) argument;

            // get cache name
            if (cacheConfiguration != null) {
                objInst.setSkyWalkingDynamicField(new EhcacheEnhanceInfo(cacheConfiguration.getName()));
            }

        } else if (argument instanceof Cache) {
            Cache cache = (Cache) argument;

            // get cache name
            if (cache != null && cache.getCacheConfiguration() != null) {
                objInst.setSkyWalkingDynamicField(new EhcacheEnhanceInfo(cache.getCacheConfiguration().getName()));
            }
        }
```

Reply:
Why? `instanceof` is still much much slower than instrumentation level match.

Reply:
ok. Thank you for your seriousness